include/openPMD/DatatypeHelpers.hpp

```diff
@@ -0,0 +1,598 @@
+            std::forward< Args >( args )... );
+    case Datatype::SHORT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< short >(
+            std::forward< Args >( args )... );
+    case Datatype::INT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< int >(
+            std::forward< Args >( args )... );
+    case Datatype::LONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< long >(
+            std::forward< Args >( args )... );
+    case Datatype::LONGLONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< long long >(
+            std::forward< Args >( args )... );
+    case Datatype::USHORT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned short >(
+            std::forward< Args >( args )... );
+    case Datatype::UINT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned int >(
+            std::forward< Args >( args )... );
+    case Datatype::ULONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned long >(
+            std::forward< Args >( args )... );
+    case Datatype::ULONGLONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned long long >(
+            std::forward< Args >( args )... );
+    case Datatype::FLOAT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< float >(
+            std::forward< Args >( args )... );
+    case Datatype::DOUBLE:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< double >(
+            std::forward< Args >( args )... );
+    case Datatype::LONG_DOUBLE:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< long double >(
+            std::forward< Args >( args )... );
+    case Datatype::CFLOAT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< std::complex< float > >(
+            std::forward< Args >( args )... );
+    case Datatype::CDOUBLE:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< std::complex< double > >(
+            std::forward< Args >( args )... );
+    // missing std::complex< long double > type in ADIOS2 v2.6.0
+    // case Datatype::CLONG_DOUBLE:
+    //     return action
+    //         .OPENPMD_TEMPLATE_OPERATOR()< std::complex< long double > >(
+    //             std::forward< Args >( args )... );
+    case Datatype::DATATYPE:
+        return detail::CallUndefinedDatatype<
+            HIGHEST_DATATYPE,
+            ReturnType,
+            Action,
+            void,
+            Args &&... >::
+            call( std::move( action ), std::forward< Args >( args )... );
+    case Datatype::UNDEFINED:
+        return detail::CallUndefinedDatatype<
+            LOWEST_DATATYPE,
+            ReturnType,
+            Action,
+            void,
+            Args &&... >::
+            call( std::move( action ), std::forward< Args >( args )... );
+    default:
+        throw std::runtime_error(
+            "Internal error: Encountered unknown datatype (switchType) ->" +
+            std::to_string( static_cast< int >( dt ) ) );
+    }
+}
+#endif // OPENPMD_SWITCHTYPE_ADIOS2
+#undef OPENPMD_TEMPLATE_OPERATOR
+}
```

Comment:
Just a missing end-of-file newline:
```suggestion
}

```